racket/collects/net/git-checkout.rkt

```diff
@@ -203,7 +203,15 @@
 (define (initial-connect transport host verify? port repo status)
   (case transport
     [(git)
-     (define-values (i o) (tcp-connect host port))
+     (define-values (i o)
+       (let ((proxy (proxy-server-for "git" host)))
+         (if proxy
+           (let ((proxy-host (cadr proxy))
+                 (proxy-port (caddr proxy)))
+             (let-values (([t:ssl-ctx t:from t:to t:abandon-p]
+                           (http-conn-CONNECT-tunnel proxy-host proxy-port host port #:ssl? #f)))
+               (values t:from t:to)))
+           (tcp-connect host port))))
```

Comment:
Pulled up into `url.rkt` as `tcp-or-tunnel-connect`
Can't go in `http-client.rkt` because it needs access to `proxy-server-for`
Shan't put it in `git-checkout.rkt` because it's too general for that